data_pipeline_api/registry/common.py

```diff
@@ -147,7 +147,8 @@ def get_remote_filesystem_and_path(
         return LocalFileSystem(**storage_options), uri.as_posix()
     elif protocol in {"http", "https", "s3"}:
         # storage_options are parameters passed to request
-        uri = urllib.parse.urljoin(uri, path)
+        path = urllib.parse.quote(path)
+        uri = '/'.join(s.strip('/') for s in [uri, path])
```

Comment:
Why are you no longer using `urljoin`? Could you use `urljoin` and then `quote`, or vice versa?

Reply:
urljoin is opinionated in ways that were not helpful to me, when I just want to be able to assume that you can do a + b and make sure there's a slash